### PR TITLE
final variables and parameters

### PR DESCRIPTION
**Pull request by thatsIch**
seeing some methods it does actually help to enforce the parameters

### CODE REVIEW

**Comment by yueh** [comment]
Not necessarily helps with enforcing parameters (or their types), but it helps the analysis as it can immensely reduce the amount of possible alternative types and thus can create better reports.

And it most likely breaks any current open PR or gets broken after merging a single one :grinning:

**Comment by thatsIch** [comment]
well this can be pulled in later on if you want. The changes are pretty straight forward for me

**Comment by yueh** [comment]
Nothing against them.

But the current sonarqube analysis might be interesting as I have added some rules yesterday :grinning:

File: src/main/java/appeng/container/implementations/ContainerCraftConfirm.java [hunk 1]
```diff
@@ -348,14 +348,14 @@ public void startJob()
 
 		if( this.result != null && !this.simulation )
 		{
-			ICraftingGrid cc = this.getGrid().getCache( ICraftingGrid.class );
-			ICraftingLink g = cc.submitJob( this.result, null, this.selectedCpu == -1 ? null : this.cpus.get( this.selectedCpu ).cpu, true, this.getActionSrc() );
+			final ICraftingGrid cc = this.getGrid().getCache( ICraftingGrid.class );
+			final ICraftingLink g = cc.submitJob( this.result, null, this.selectedCpu == -1 ? null : this.cpus.get( this.selectedCpu ).cpu, true, this.getActionSrc() );
 			this.autoStart = false;
 			if( g != null && originalGui != null && this.openContext != null )
```
Comment (orod-org) [inline src/main/java/appeng/container/implementations/ContainerCraftConfirm.java]: MINOR Unwritten public or protected field: appeng.container.AEBaseContainer.openContext [rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3AUWF_UNWRITTEN_PUBLIC_OR_PROTECTED_FIELD)

**Comment by orod-org** [comment]
SonarQube analysis reported 2 issues:
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* INFO [Unread public/protected field: appeng.core.AEConfig.minMeteoriteDistanceSq](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/5e5c5ce43cdee0691b3eae8e2110043c61b1fae1/src/main/java/appeng/core/AEConfig.java#L75) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3AURF_UNREAD_PUBLIC_OR_PROTECTED_FIELD)

**Comment by orod-org** [comment]
SonarQube analysis reported 2 issues:
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* INFO [Unread public/protected field: appeng.core.AEConfig.minMeteoriteDistanceSq](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/410d2f1e0da6fc23a54fcf2286cad73024634f45/src/main/java/appeng/core/AEConfig.java#L75) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3AURF_UNREAD_PUBLIC_OR_PROTECTED_FIELD)

**Comment by thatsIch** [comment]
only activated like 20-30% of the check styles and already 300.000 problems :laughing:

**Pull request merged**